src/components/Input.react.js

```diff
@@ -65,22 +65,27 @@ export default class Input extends Component {
                     if (setProps) {
                         const castValue =
                             type === 'number' ? Number(value) : value;
```

Comment:
For consistency, the way the `payload` is set up should probably be the same in the `onBlur` method as well as in the `onKeyPress` method. I might prefer the way it's done on line 86 - the `castvalue` const also shouldn't have to be defined if it's not being used here.